Lib/asyncio/proactor_events.py

```diff
@@ -425,6 +444,134 @@ def _pipe_closed(self, fut):
             self.close()
 
 
+class _ProactorDatagramTransport(_ProactorBasePipeTransport):
+    max_size = 256 * 1024
+    def __init__(self, loop, sock, protocol, address=None,
+                 waiter=None, extra=None):
+        self._address = address
+        self._empty_waiter = None
+        # We don't need to call _protocol.connection_made() since our base
+        # constructor does it for us.
+        super().__init__(loop, sock, protocol, waiter=waiter, extra=extra)
+
+        # The base constructor sets _buffer = None, so we set it here
+        self._buffer = collections.deque()
+        self._loop.call_soon(self._loop_reading)
+
+    def _set_extra(self, sock):
+        _set_socket_extra(self, sock)
+
+    def get_write_buffer_size(self):
+        return sum(len(data) for data, _ in self._buffer)
+
+    def abort(self):
+        self._force_close(None)
+
+    def sendto(self, data, addr=None):
+        if not isinstance(data, (bytes, bytearray, memoryview)):
+            raise TypeError('data argument must be byte-ish (%r)',
```

Comment:
```suggestion
            raise TypeError('data argument must be bytes-like object (%r)',
```

`byte-ish` could be rephrased as `bytes-like object` which is a more common term used in other places where same check is made.

```
$ rg -A 2 'not isinstance\(data, \(bytes, bytearray, memoryview\)'

Lib/asynchat.py
193:        if not isinstance(data, (bytes, bytearray, memoryview)):
194-            raise TypeError('data argument must be byte-ish (%r)',
195-                            type(data))

Lib/asyncio/proactor_events.py
307:        if not isinstance(data, (bytes, bytearray, memoryview)):
308-            raise TypeError(
309-                f"data argument must be a bytes-like object, "

Lib/asyncio/sslproto.py
382:        if not isinstance(data, (bytes, bytearray, memoryview)):
383-            raise TypeError(f"data: expecting a bytes-like instance, "
384-                            f"got {type(data).__name__}")

Lib/asyncio/selector_events.py
852:        if not isinstance(data, (bytes, bytearray, memoryview)):
853-            raise TypeError(f'data argument must be a bytes-like object, '
854-                            f'not {type(data).__name__!r}')
--
979:        if not isinstance(data, (bytes, bytearray, memoryview)):
980-            raise TypeError(f'data argument must be a bytes-like object, '
981-                            f'not {type(data).__name__!r}')
```

Reply:
Good catch!
`byte-ish` was living in asyncio from the very beginning and for 5 year.